snapcraft/internal/remote_build/errors.py

```diff
@@ -126,3 +129,18 @@ class AcceptPublicUploadError(RemoteBuildBaseError):
         "Remote build needs explicit acknowledgement that data sent to build servers "
         "is public.\nIn non-interactive runs, please use option --accept-public-upload."
     )
+
+
+class LaunchpadGitPushError(SnapcraftException):
+    def __init__(self, *, command: str, exit_code: int) -> None:
+        self._command = command
+        self._exit_code = exit_code
+
+    def get_brief(self) -> str:
+        return "Failed to push sources to launchpad."
+
+    def get_details(self) -> str:
+        return f"Command {self._command!r} failed with exit code {self._exit_code!r}."
+
+    def get_resolution(self) -> str:
+        return "Verify connectivity to https://git.launchpad.net and retry build."
```

Comment:
oh, maybe not, full url in brief is not needed as it is clarified here.

Reply:
The stripped url is also provided in the command (details)